hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -120,6 +122,7 @@ public Set<ConfigOption<?>> optionalOptions() {
    */
   private void sanityCheck(Configuration conf, ResolvedSchema schema) {
     List<String> fields = schema.getColumnNames();
+    Schema inferredSchema = AvroSchemaConverter.convertToSchema(schema.toPhysicalRowDataType().notNull().getLogicalType());
 
```

Comment:
There is no need to convert the `ResolvedSchema` as an avro schema for validation, the `ResolvedSchema#getColumnDataTypes` can fetch the data type of each field.

Also we need to fix the `RowDataKeyGen#getRecordKey` for nested primary keys.

Reply:
Sure, I will take a look. 

The main reasons for doing this is:
1. `AvroSchemaConverter.convertToSchema` was already imported and used somewhere else in the code so, just reuse
2. Convert it to an AvroSchema so that the helper functions can be written in HoodieAvroUtils, where the validation for creation of tables using the Spark as entrypoint can be reused.

Let me try to see if we can use the `ResolvedSchema` instead, will get back to you.